File: base/interfaces/identity.go
package interfaces

import (
	"base/constants"
)

//||------------------------------------------------------------------------------------------------||
//|| Identity (Epic Verified Profile)
//||------------------------------------------------------------------------------------------------||

type Identity struct {
	Email      string                       `json:"email,omitempty"`
	Age        string                       `json:"age,omitempty"`
	Phone      string                       `json:"phone,omitempty"`
	Address    string                       `json:"address,omitempty"`
	CreditCard string                       `json:"creditCard,omitempty"`
	Usernames  map[int64]string             `json:"usernames,omitempty"`
	Approved   []constants.VerificationType `json:"approved,omitempty"`
}

File: ui/src/interfaces/identity/identity.ts

//||------------------------------------------------------------------------------------------------||
//|| Interfaces : Identity
//|| Identity Verification Data
//||------------------------------------------------------------------------------------------------||

import { VerificationTypes }                                from "../models/model.verify";
import { CreditCard }                                       from "../base/transaction";
import { Address }                                          from "../base/geo";
import { DOB, EmailAddress, PhoneNumber, IDCard, Username}  from "../base/user";

//||------------------------------------------------------------------------------------------------||
//|| Identity (Epic Verified Profile) - TypeScript Interfaces
//||------------------------------------------------------------------------------------------------||

export interface IdentityUsername {
      idSite            : number;
      username          : string;
      verification      : string;
}

export interface IdentityRecord {
      display           : string;
      verification      : string;
}

export interface Identity {
      email?            : IdentityRecord;
      age?              : IdentityRecord;
      phone?            : IdentityRecord;
      address?          : IdentityRecord;
      creditCard?       : IdentityRecord;
      idCard?           : IdentityRecord;
      usernames?        : Record<string, IdentityUsername>;
      approved?         : string[];
      verified          : boolean;
      verifiedAge?      : number;
}
